tldw_Server_API/tests/AuthNZ/unit/test_auth_endpoints_extended.py

```
@@ -265,6 +266,145 @@ async def _governor(_request):
+    assert excinfo.value.detail == "Admin reauthentication is only available to administrators"
+
+
+@pytest.mark.asyncio
+async def test_verify_magic_link_rejects_admin_reauth_purpose(monkeypatch):
+    reset_settings()
+    monkeypatch.setenv("AUTH_MODE", "multi_user")
+
+    import tldw_Server_API.app.api.v1.endpoints.auth as auth
+
+    class _StubJWT:
+        async def verify_token_async(self, token: str, token_type: str | None = None):
```

Ruff (0.15.6)

[warning] Missing return type annotation for private function `verify_token_async`

(ANN202)